backend/core/Extensions/EnumerableExtensions.cs

```diff
@@ -39,6 +40,36 @@ public static void ForEach<T>(this IEnumerable<T> items, Action<T, int> action)
             }
         }
 
+        /// <summary>
+        /// Iterates through the enumerable collection and performs the specified action.
+        /// </summary>
+        /// <param name="items"></param>
+        /// <param name="action"></param>
+        /// <typeparam name="T"></typeparam>
+        public static async Task ForEachAsync<T>(this IEnumerable<T> items, Func<T, Task> action)
```

Comment:
This will invoke all of these actions serially correct? to we want a Parallel version?

Reply:
Good suggestion, I'll update.